### PR TITLE
Add global const fnv_hash function

### PR DESCRIPTION
**Pull request by stepancheg**
Sometimes need to compute hash of a string at compile time.

### CODE REVIEW

File: lib.rs [hunk 1]
```diff
@@ -136,6 +136,20 @@ pub type FnvHashMap<K, V> = HashMap<K, V, FnvBuildHasher>;
 pub type FnvHashSet<T> = HashSet<T, FnvBuildHasher>;
 
 
+/// Const version of FNV hash.
+#[inline]
+pub const fn fnv_hash(bytes: &[u8]) -> u64 {
+    let mut hash = 0xcbf29ce484222325;
```
Comment (Manishearth) [inline lib.rs:142]: nit: this should use some shared constant

File: lib.rs [hunk 2]
```diff
@@ -136,6 +136,20 @@ pub type FnvHashMap<K, V> = HashMap<K, V, FnvBuildHasher>;
...
+    let mut i = 0;
+    while i < bytes.len() {
+        hash = hash ^ (bytes[i] as u64);
+        hash = hash.wrapping_mul(0x100000001b3);
```
Comment (Manishearth) [inline lib.rs:146]: nit: this too

**Review by Manishearth — COMMENTED** [review]
I think this is probably a good idea, but we should avoid duplicating code as much as possible

(I'm aware that duplicating the implementation code is probably a good idea due to perf impacts)

**Comment by stepancheg** [comment]
Missed notification from this PR. Updated now.

**Review by Manishearth — APPROVED** [review]

**Comment by Manishearth** [comment]
We should probably switch over to GHA

**Pull request merged**